src/global_timestamp_reader.cpp

```diff
@@ -111,9 +112,26 @@ namespace librealsense
 
     void time_diff_keeper::start()
     {
+        _users_count++;
+        LOG_DEBUG("time_diff_keeper::start: _users_count = " << _users_count);
         _active_object.start();
     }
 
+    void time_diff_keeper::stop()
+    {
+        if (_users_count <= 0)
```

Comment:
Couple of options - 
a. Use regular uint instead of an atomic and wrap read/writes with a mutex
b. Use atomic compare-exchange pattern